addons/html_builder/static/src/plugins/image_gallery_option_plugin.js

```diff
@@ -401,6 +403,29 @@ class ImageGalleryOption extends Plugin {
         );
     }
 
+    onRemove(elementToRemove) {
+        // If the removed element is an image from a gallery, set the flag to true
+        // and store the gallery element for afterRemove
+        if (!this.isGalleryImage) {
+            this.isGalleryImage = false;
+        }
+        if (elementToRemove.matches(".s_image_gallery img")) {
+            this.isGalleryImage = true;
+            this.editingGalleryElement = elementToRemove.closest(".s_image_gallery");
```

Comment:
Can we get rid of `isGalleryImage` and only use the gallery element variable ? (if you have a better name don't hesitate)
```suggestion
            this.removedImageGallery = elementToRemove.closest(".s_image_gallery");
```